advntr/vntr_finder.py

```diff
@@ -366,23 +366,52 @@ def check_if_pacbio_read_spans_vntr(self, sema, read, length_distribution, spann
         sema.release()
 
     def check_if_pacbio_mapped_read_spans_vntr(self, sema, read, length_distribution, spanning_reads):
-        flanking_region_size = 100
-        region_start = self.reference_vntr.start_point - flanking_region_size
-        region_end = self.reference_vntr.start_point + self.reference_vntr.get_length()
-        if read.get_reference_positions()[0] < region_start and read.get_reference_positions()[-1] > region_end:
+        hmm_flanking_region_size = 100
+        min_flanking_bp = 10
+        vntr_start = self.reference_vntr.start_point
+        vntr_end = self.reference_vntr.start_point + self.reference_vntr.get_length()
+
+        region_start = vntr_start - hmm_flanking_region_size
+
+        first_aligned_position = read.get_reference_positions()[0]
+        last_aligned_position = read.get_reference_positions()[-1]
+        if first_aligned_position <= vntr_start - min_flanking_bp and vntr_end + min_flanking_bp < last_aligned_position:
```

Comment:
Could you please add a check before this line to make sure `first_aligned_position` and `last_aligne_position` are not `None`?

Reply:
They can never be `None` without `full_length=True` option because they are mapped reads.

Reply:
Oh I get it. Makes sense.